js/releases.js

```diff
@@ -0,0 +1,127 @@
+    const asset = filter ? release.assets.find(item => item.name.indexOf(filter) > -1) : undefined;
+    const result = document.createElement("li");
+    if (asset) {
+        const link = document.createElement("a");
+        link.innerText = name;
+        link.target = "_blank";
+        link.href = asset.browser_download_url;
+        result.appendChild(link);
+    } else {
+        result.innerText = name;
+        result.style.textDecoration = "line-through";
+    }
+    return result;
+}
+
+function desktopDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Linux", "-linux"));
+    list.appendChild(downloadLink(release, "Mac OS", "-macos"));
+    list.appendChild(downloadLink(release, "Windows (32-bit)", "-windows-x86_32"));
+    list.appendChild(downloadLink(release, "Windows (64-bit)", "-windows-x86_64"));
+
+    result.appendChild(list);
+    return result;
+}
+
+function browserDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Firefox", "-firefox-unsigned"));
+    list.appendChild(downloadLink(release, "Chrome / Edge", "-extension."));
+    list.appendChild(downloadLink(release, "Safari", false));
+
+    result.appendChild(list);
+    return result;
+}
+
+function webDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Self Hosted", "-selfhosted"));
+
+    result.appendChild(list);
+    return result;
+}
+
+function updateTable(table_id, releases, limit, withDate, withName) {
+    const table = document.getElementById(table_id);
+    const entries = document.getElementById(table_id + "_entries");
+    if (releases.length === 0) {
+        table.style.display = "none";
+        return;
+    }
+    table.style.display = "table";
+    while (entries.firstChild) {
+        entries.removeChild(entries.firstChild);
+    }
+    for (let i = 0; i < Math.min(releases.length, limit); i++) {
+        const release = releases[i];
+        const row = document.createElement("tr");
+
+        if (withName) row.appendChild(releaseName(release));
+        if (withDate) row.appendChild(publishDate(release));
+        row.appendChild(desktopDownloads(release));
+        row.appendChild(browserDownloads(release));
+        row.appendChild(webDownloads(release));
+
+        entries.appendChild(row);
+    }
+}
+
+(async () => {
+    const { data: releases } = await octokit.rest.repos.listReleases({
+        owner: "ruffle-rs",
+        repo: "ruffle",
+    });
+
+    const updateReleases = () => {
+        updateTable("downloads_table", releases.filter(r => !r.draft), 1, false, false);

```

Comment:
This defeats the purpose of "Latest Nightly Release" and is actually functionally different to how it currently behaves. This just displays all the possibly non-existent assets of the latest release. The current way "Latest Nightly Release" works is that it shows the most recent existing asset of each type, so if the web build fails one day, it shows the previous day's web builds but today's desktop builds. It was added in #63.

Reply:
I'm not sure "Latest Nightly Release" is that useful anymore. However, it's been helpful when we've had CI issues in the past. I wouldn't mind it being removed personally though. Maybe @midgleyc, as the one who added it, has further opinions.

Keeping it's current functionality will probably involve filtering the releases array with lots of ES6.

Reply:
I added it back when Firefox wasn't routinely updating in the extension store, so the nightly release was practically the only was to use it.

Now that the extension is updating regularly (and stably), there's little need for this I agree.